data-access/mandataris-download.ts

```diff
@@ -0,0 +1,243 @@
+import { query, sparqlEscapeString } from 'mu';
+import { getSparqlResults } from '../util/sparql-result';
+import { sparqlEscapeDateTime, sparqlEscapeUri } from '../util/mu';
+
+export const downloadMandatarissen = {
+  getWithFilters,
+  getPropertiesOfMandatarissen,
+};
+
+async function getWithFilters(filters) {
+  const {
+    bestuursperiodeId,
+    bestuursorgaanId,
+    persoonIds,
+    fractieIds,
+    hasFilterOnOnafhankelijkeFractie,
+    hasFilterOnNietBeschikbareFractie,
+    bestuursFunctieCodeIds,
+  } = filters;
+  let bestuursorgaanInTijdFilter: string | null = null;
+  let onlyActiveFilter: string | null = null;
+  let persoonFilter: string | null = null;
+  let fractieFilter: string | null = null;
+  let fractieOnafhankelijkFilter: string | null = null;
+  let fractieOOBFilter: string | null = null;
+  let mandaatTypeFilter: string | null = null;
+
+  if (bestuursorgaanId) {
+    bestuursorgaanInTijdFilter = `
+      ?mandaat ^org:hasPost ?bestuursorgaanInTijd.
+      ?bestuursorgaanInTijd mu:uuid ${sparqlEscapeString(bestuursorgaanId)}.
+    `;
+  }
+  if (filters.onlyShowActive) {
+    const escapedTodaysDate = sparqlEscapeDateTime(new Date());
+    onlyActiveFilter = `
+      ?mandataris mandaat:einde ?einde.
+
+      ?bestuursorgaanInTijd mandaat:bindingStart ?startBestuursorgaan. 
+      ?bestuursorgaanInTijd mandaat:bindingEinde ?eindeBestuursorgaan. 
+
+      FILTER (
+        ?einde <= ?eindeBestuursorgaan &&
+        ?einde >= ?startBestuursorgaan &&
+        ?einde >= ${escapedTodaysDate}
+      )
+    `;
+  }
+
+  if (persoonIds.length >= 1) {
+    const idValues = persoonIds.map((id) => sparqlEscapeString(id)).join(' ');
+    persoonFilter = `
+      VALUES ?persoonId { ${idValues} }
+
+      ?mandataris mandaat:isBestuurlijkeAliasVan ?persoon.
+      ?persoon mu:uuid ?persoonId.
+    `;
+  }
+
+  if (fractieIds.length >= 1) {
+    const idValues = fractieIds.map((id) => sparqlEscapeString(id)).join(' ');
+    fractieFilter = `
+      {
+        VALUES ?fractieId { ${idValues} }
+
+        ?mandataris org:hasMembership ?lidmaatschap.
+        ?lidmaatschap org:organisation ?fractie.
+        ?fractie mu:uuid ?fractieId.  
+      }
+    `;
+  }
+
+  if (hasFilterOnOnafhankelijkeFractie) {
+    if (fractieIds.length >= 1) {
+      fractieOnafhankelijkFilter = `
+        UNION {
+          ?fractie ext:isFractietype fractieType:Onafhankelijk.
+        }
+      `;
+    } else {
+      fractieOnafhankelijkFilter = `
+      ?mandataris org:hasMembership ?lidmaatschap.
+      ?lidmaatschap org:organisation ?fractie.
+      ?fractie ext:isFractietype fractieType:Onafhankelijk.
+      `;
+    }
+  }
+  if (hasFilterOnNietBeschikbareFractie) {
+    fractieOOBFilter = `
+      FILTER NOT EXISTS {
+        ?mandataris org:hasMembership ?lidmaatschap.
+      }
+    `;
+  }
```

Comment:
It filters it but it sometimes returns no results when there is one

Reply:
i would try adding some more predicates to the not exist, including the link to fractie and the type
maybe this is a predicate that resources didn't remove when the instance got destroyed (or something) because it points from mandataris to membership